backend/api/Services/AutoScheduleService.cs

```diff
@@ -0,0 +1,81 @@
+using System.Text.Json;
+using Api.Database.Models;
+
+namespace Api.Services
+{
+    public interface IAutoScheduleService
+    {
+        public Dictionary<TimeOnly, string> DeserializeAutoScheduleJobs(
+            MissionDefinition missionDefinition,
+            ILogger logger
+        );
+        public void ReportSkipAutoScheduleToSignalR(
+            string message,
+            MissionDefinition missionDefinition
+        );
+        public void ReportAutoScheduleFailToSignalR(
+            string message,
+            MissionDefinition missionDefinition
+        );
+    }
+
+    public class AutoScheduleService(
+        ILogger<AutoScheduleService> logger,
+        ISignalRService signalRService
+    ) : IAutoScheduleService
+    {
+        public Dictionary<TimeOnly, string> DeserializeAutoScheduleJobs(
+            MissionDefinition missionDefinition,
+            ILogger logger
+        )
```

Comment:
Why send in a logger and not just use the logger already defined in the service?